src/mscorlib/src/System/Span.cs

```diff
@@ -289,6 +289,25 @@ public void Set(ReadOnlySpan<T> values)
 
             SpanHelper.CopyTo<T>(ref GetRawPointer(), ref values.GetRawPointer(), values.Length);
         }
+
+        /// <summary>
+        /// Fills the contents of this span with the value.
+        /// </summary>
+        /// <param name="value">The value to fill the span content with.</param>
+        public void Fill(T value)
+        {
+            if (_length > 0)
+                SpanHelper.Fill<T>(ref GetRawPointer(), value, _length);
+        }
+
+        /// <summary>
+        /// Fills the contents of this span with the default value.
+        /// </summary>
+        public void Clear()
+        {
+            if (_length > 0)
+                SpanHelper.Fill<T>(ref GetRawPointer(), default(T), _length);
```

Comment:
References are special because updating a reference in memory requires a GC write barrier. But if the reference you are writing to memory is `null` then no write barrier is needed, you can simply write a 0 and call it a day.

That said, you may still need to account for references depending on the way zeroes are written. For example, you can't just call `Buffer.ZeroMemory` because that one writes one byte at a time and if the GC stops you in the middle then it may end up seeing a partially updated and thus corrupted reference.

But typical clear implementations use large aligned writes so in general this shouldn't be an issue.